src/astUtils/visitors.ts

```diff
@@ -11,7 +11,7 @@ import type { AstEditor } from './AstEditor';
  */
 export function walkStatements(
     statement: Statement,
-    visitor: (statement: Statement, parent?: Statement) => Statement | void,
+    visitor: (statement: Statement, parent?: Statement, owningObject?: any, key?: any) => Statement | void,
```

Comment:
@chrisdp what do you think of the name `owningObject`? Do you hate it? love it? have an alternative?